src/data/finsupp/basic.lean

```diff
@@ -1825,6 +1829,26 @@ begin
     apply h_cases (hl $ finset.mem_coe.2 $ mem_support_iff.2 $ λ h, h_contr h.symm) }
 end
 
+variable {f}
+
+@[simp] lemma comap_domain_zero :
+  comap_domain f (0 : β →₀ M) (hf.inj_on _) = (0 : α →₀ M) :=
+by { ext, simp only [comap_domain_apply, coe_zero, pi.zero_apply] }
+
+lemma comap_domain_add :
```

Comment:
Can you add the smul lemma too?

Reply:
You mean that I should assume that `α β` have a scalar action and prove "linearity"?

Reply:
Ah, maybe you meant using multiplication by `nat`s?  I just added this lemma.

Reply:
Yes, when I said `α β` I really meant `M`!  I'll get on with it!